src/System.Private.CoreLib/src/System/Runtime/CompilerServices/ClassConstructorRunner.cs

```diff
@@ -30,11 +30,29 @@ internal static partial class ClassConstructorRunner
         //
         // No attempt is made to detect or break deadlocks due to other synchronization mechanisms.
         //==============================================================================================================
-#if !CORERT // CORERT-TODO: Use full cctor helper
+#if !CORERT
         [RuntimeExport("CheckStaticClassConstruction")]
-#endif
         public static unsafe void* EnsureClassConstructorRun(void* returnValue, StaticClassConstructionContext* pContext)
         {
+            EnsureClassConstructorRun(pContext);
+            return returnValue;
+        }
+#else
+        private unsafe static object CheckStaticClassConstructionReturnGCStaticBase(StaticClassConstructionContext* context, object gcStaticBase)
+        {
```

Comment:
The fast path is multiple jump and calls now (R2R helper -> CheckStaticClassConstructionReturnGCStaticBase -> EnsureClassConstructorRun). This will need more fine-tuning, e.g. inline the fast path into R2R helper.